src/endpoint/s3/ops/s3_get_object.js

```diff
@@ -52,14 +52,19 @@ async function get_object(req, res) {
     if (md_params.get_from_cache) {
         params.get_from_cache = true;
     }
-    if (part_number) {
```

Comment:
I don't think this code should be removed for all cases.   
notice that `part_number` is checked in `namespace_cache.js` both for `read_object_md` and `read_object_stream`. 
since `part_number` was sent in `md_params` above, you will not get the start and end in object_md, but you also don't send the `part_number` now so you will get the wrong result

Reply:
OK, restoring